Beam/resources/assets/js/remplib.js

```diff
@@ -657,7 +662,7 @@ remplib = typeof(remplib) === 'undefined' ? {} : remplib;
             }
 
 
-            const params = {
+            var params = {
                 "action": "progress",
```

Comment:
@rootpd Pri testovani mi backend vratil chybu ze action moze byt len "load" alebo "timespent". Je to vec ktoru my riesit nemusime ?

Reply:
Mozem vidiet cely payload? Mas urcite aktualnu verziu trackeru? https://github.com/remp2020/remp/releases/tag/0.5.2

```
➜  remp git:(master) ✗ curl -v -X POST \
  http://tracker.beam.remp.press/track/pageview \
  -H 'Content-Type: application/json' \
  -H 'Origin: http://campaign.remp.press' \
  -d '{
    "article": {
        "id": "9900222",
        "author_id": "autoresku blesku 222",
        "category": "kategoria 222",
        "locked": true,
        "tags": ["tag1", "tag2", "tag3"],
        "variants": {}
    },
    "action": "progress",
    "progress": {
        "page_ratio": 0.203,
        "article_ratio": 0.95
    },
    "system": {
        "property_token": "1a8feb16-3e30-4f9b-bf74-20037ea8505a",
        "time": "2019-01-25T15:21:27.293Z"
    },
    "user": {
        "browser_id": "0fa7e641-7573-4ccd-90da-c7b902b08755",
        "subscriber": true,
        "referer": "",
        "adblock": true,
        "window_height": 1243,
        "window_width": 1603,
        "cookies": true,
        "websockets": true,
        "source": {
            "ref": "direct"
        },
        "remp_session_id": "2d06f9c3-2e73-4c53-9407-a5958876ef7b",
        "remp_pageview_id": "34eeae3b-0223-422e-9175-f8be71a3c6ac"
    }
}'
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to tracker.beam.remp.press (127.0.0.1) port 80 (#0)
> POST /track/pageview HTTP/1.1
> Host: tracker.beam.remp.press
> User-Agent: curl/7.61.1
> Accept: */*
> Content-Type: application/json
> Origin: http://campaign.remp.press
> Content-Length: 930
> 
* upload completely sent off: 930 out of 930 bytes
< HTTP/1.1 202 Accepted
< Server: nginx/1.14.2
< Date: Mon, 13 May 2019 07:36:28 GMT
< Content-Length: 0
< Connection: keep-alive
< Access-Control-Allow-Credentials: false
< Access-Control-Allow-Origin: http://campaign.remp.press
< 
* Connection #0 to host tracker.beam.remp.press left intact
```

Reply:
Myslim, ze mame najaktualnejsiu verziu. Dnes sme to skusali aj s tou najaktualnejsou. Tu je teda cely payload. 

`curl 'http://tracker.beam.remp.press:8000/track/pageview' -H 'Referer: http://beam.remp.press:8000/test.html' -H 'Origin: http://beam.remp.press:8000' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.86 Safari/537.36' -H 'Content-Type: application/json' --data-binary '{"action":"progress","progress":{"page_ratio":0.48602673147023084,"unload":false,"article_ratio":0.2549728752260398},"system":{"property_token":"8f406aaa-4e70-35cc-a054-903acd1ca587","time":"2019-05-14T14:48:46.165Z"},"user":{"id":"1","browser_id":"d6f5c542-fdd7-4aff-852b-012509a0b4a1","subscriber":true,"url":"http://beam.remp.press:8000/test.html","referer":"","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.86 Safari/537.36","adblock":false,"window_height":856,"window_width":1409,"cookies":true,"websockets":true,"source":{},"remp_session_id":"c5db358b-e51f-4ee1-9f5a-05dcf647629b","remp_pageview_id":"e07a962a-70e1-4075-9d47-ee887759e5bc"}}' --compressed`


Vracia to toto: 
`{"id":"PP5mwimP","code":"bad_request","status":400,"detail":"[Eif05XET] 400 invalid_request: value of response.action must be one of \"load\", \"timespent\" but got value \"progress\", attribute: response.action, value: progress, expected: \"load\", \"timespent\""}
`

Reply:
Rovnaky payload vracia u mna 202 accepted (vymenil som len `property_token`, ten mame logicky iny). Ak si teda v dockri, spusti v roote projektu:

```
make docker-build
docker-compose stop beam_tracker
docker-compose build beam_tracker
docker-compose up -d beam_tracker
```

Ak si mimo dockeru, stiahni si aktualnu binarku tu: https://github.com/remp2020/remp/releases/tag/0.5.3